### PR TITLE
Prevent common properties in Case Search properties and filters

### PR DESCRIPTION
**Pull request by sravfeyn**
## Summary
Adds a UI constraint to not let users create case search properties and filters on same property

https://dimagi-dev.atlassian.net/browse/USH-1147

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### QA Plan

Not necessary as it's a simple UI change

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/app_manager/static/app_manager/js/details/case_claim.js [hunk 1]
```diff
@@ -361,6 +361,21 @@ hqDefine("app_manager/js/details/case_claim", function () {
             );
         };
 
+        self.commonProperties = ko.computed(function () {
```
Comment (orangejenny) [inline corehq/apps/app_manager/static/app_manager/js/details/case_claim.js:364]: [intersect](https://underscorejs.org/#intersection) might be useful in this function

File: corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html [hunk 2]
```diff
@@ -55,8 +55,12 @@ <h4 class="panel-title panel-title-nolink">{% trans "Default Search Properties"
           </thead>
           <tbody data-bind="foreach: defaultProperties">
           <tr>
-            <td class="col-sm-4">
+            <td class="col-sm-4" data-bind="css: {'has-error': $parent.commonProperties().indexOf(ko.unwrap(property())) !== -1}">
```
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html:58]: This expression is a bit unwieldy, could you extract it as a computed observable with a clear name?

**Review by orangejenny — COMMENTED** [review]
Just minor comments.

File: corehq/apps/app_manager/static/app_manager/js/details/case_claim.js (hunk 1 above)
Comment (sravfeyn) [reply corehq/apps/app_manager/static/app_manager/js/details/case_claim.js:364]: The objects inside have different structure, we have to intersect by `name` attr of one with the `'property` attr of other, so I went for the manual method here.

File: corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html (hunk 2 above)
Comment (sravfeyn) [reply corehq/apps/app_manager/templates/app_manager/partials/modules/case_search_properties.html]: True, I have updated it.

**Review by orangejenny — APPROVED** [review]

**Pull request merged**